great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -1038,3 +1076,46 @@ def add_query_asset(
             batch_metadata=batch_metadata or {},
         )
         return self._add_asset(asset)
+
+
+def _verify_table_name_exists_and_get_normalized_typed_name_map(
+    name: str,
+    typed_names: List[str | sqlalchemy.quoted_name],
+    error_message_template: str = 'Error: The table "{table_name:s}" does not exist.',
+    verify_only: bool = False,
+) -> Tuple[str, str | sqlalchemy.quoted_name] | None:
+    """
+    Verifies that table name exists.
+
+    Args:
+        name: string-valued name
+        typed_names: Properly typed names (e.g., output of "inspector.get_table_names()" SQLAlchemy method call)
+        verify_only: Perform verification only (do not return normalized typed name map)
+        error_message_template: String template to output error message if name cannot be found in typed_names list
+
+    Returns:
+        Single tuple holding mapping from string-valued name to typed name; None if "verify_only" is set.
+    """
+
+    def _get_normalized_table_name_mapping_if_exists(
+        name: str,
+    ) -> Tuple[str, str | sqlalchemy.quoted_name] | None:
+        typed_name_cursor: str | sqlalchemy.quoted_name
+        for typed_name_cursor in typed_names:
+            if (
+                (type(typed_name_cursor) == str)
+                and (name.casefold() == typed_name_cursor.casefold())
+                or (name == str(typed_name_cursor))
+            ):
```

Comment:
why `type` instead of `isinstance`?

Reply:
@cdkini Great question: the `isinstance` did not work for me because `sqlalchemy.quoted_name` extends `str`, so I had to be precise.  Thanks.